### ISSUE TITLE
Autostorage and Autosell don't work together

### ISSUE THREAD

**Issue by ncplay91** [OpenKore/openkore#1371]
Please refer to the following resources for help using OpenKore:
 faqs - http://openkore.com/index.php/Frequently_Asked_Questions (such as "Packet Tokenizer")
 forum - http://forums.openkore.com
 documentation wiki - http://openkore.com/index.php/Main_Page
 download new openkore - https://github.com/OpenKore/openkore/archive/master.zip
* Before submitting an issue, ensure you are on the latest release
* Check if an issue already exists for your request
* Please post in English if possible
* Please use the issue template, please post infornations about the server and your configuration

###### Summary:
Bot sells everything in inventory even though items_control.txt has been set.
Re;Start server
###### Affected configuration(s)/ file(s):
```Include relevant code snippets and source code references```
config.txt

storageAuto 1
storageAuto_npc comodo 195 150
storageAuto_distance 1
storageAuto_npc_type 1
storageAuto_npc_steps c c r1 n
storageAuto_password
storageEncryptKey
storageAuto_keepOpen 0
storageAuto_useChatCommand
storageAuto_notAfterDeath
relogAfterStorage 0
minStorageZeny 0

npcTimeResponse
npcWrongStepsMethod 0

getAuto Awakening Potion {
	minAmount 0
	maxAmount 50
	passive 1
	dcOnEmpty 0
}

sellAuto 1
sellAuto_npc comodo 225 164
sellAuto_standpoint
sellAuto_distance 2
sellAuto_npc_steps

items_control.txt
all 0 0 1

Hydra Card 0 1 0

###### Impact:
Bot opens storage but doesn't store anything instead, bot sells everything.

###### Expected Behavior:
bot stores some items and sell some items.

###### Actual Behavior:
bot opens storage and close it immediately without performing any other storing action then proceed to sell all items in the inventory.

###### Affected configuration(s)/ file(s):
```Include relevant code snippets and source code references```

**Comment by ncplay91** [comment]
btw the server is restart..

**Comment by alisonrag** [comment]
did you readed the documentation?
you cant do it:
items_control.txt
all 0 0 1
and spect that openkore store everything

download the latest version of openkore:

**Comment by ncplay91** [comment]
mm... what I want to do is to sell everything except hydra card..
so in items_control.txt I set,
all 0 0 1
Hydra Card 0 1 0
I think it should be fine.. but I dont know somehow it sells everything and ignore that 0 1 0 I specified.

**Comment by alisonrag** [comment]
try the id of hydra card:

```
all 0 0 1 0 0
4035 0 1 0 0 0
```

**Comment by ncplay91** [comment]
Testing... this is gonna take forever to get hydra card

**Comment by alisonrag** [comment]
delete everything in items_control and test it:
```
4035 0 1 0 0 0
all 0 0 1 0 0
```

**Comment by alisonrag** [comment]
any update?

**Comment by ncplay91** [comment]
It's still doesn't work..
I tried changing hydra with other items.. and Kore still sells everything.. this is weird..

**Comment by c4c1n6kr3m1** [comment]
did you do @alisonrag told you ?
clear all content inside items_control.txt
than start write
4035 0 1 0 0 0
all 0 0 1 0 0

**Comment by fadreus** [comment]
@ncplay91 Upload your `config.txt` and `items_control.txt` here.

**Comment by ncplay91** [comment]
Here's the config:

storageAuto 1
storageAuto_npc comodo 195 150
storageAuto_distance 1
storageAuto_npc_type 1
storageAuto_npc_steps c c r1 n
storageAuto_password
storageEncryptKey
storageAuto_keepOpen 0
storageAuto_useChatCommand
storageAuto_notAfterDeath
relogAfterStorage 0
minStorageZeny 0

npcTimeResponse
npcWrongStepsMethod 0

getAuto Awakening Potion {
	minAmount 1
	maxAmount 30
	passive 1
	dcOnEmpty 0
}

sellAuto 1
sellAuto_npc cmd_in01 79 182
sellAuto_standpoint
sellAuto_distance 5
sellAuto_npc_steps

and this is the items_control.txt:

all 0 0 1

4035 0 1 0 0 0 (-> this is Hydra Card like @alisonrag suggested)
Hydra Card 0 1 0
Awakening Potion 30 1 0

The funny thing is that.. Kore sells everything in my inventory except the 30 awakening potions it took from kafra..
and when I set Hydra Card 100 1 0... Kore still sells my Hydra Card T.T.. I was like.. seriously!?

**Comment by sctnightcore** [comment]
> sellAuto_npc_steps

**Comment by c4c1n6kr3m1** [comment]
this is rough, if you want to try, i haven't test, my bot is years late
src\AI\CoreLogic.pm
find this line
```next if ($item->{equipped} || !$item->{sellable});```
add this code after that one
```next if ($item->{name} =~ / Card$/i);```

**Comment by fadreus** [comment]
@ncplay91 Lol. Upload bro, not copy paste.

**Comment by alisonrag** [comment]
you are the one here with this problem.

can you download the latest version and clean make only the necessary configuration?

**Comment by ncplay91** [comment]
@alisonrag I've already downloaded the latest openkore.. and using the same config as above..
somehow the bot still sells everything despite the items_control.txt has been properly set up..
how is this happening?

**Comment by fadreus** [comment]
Just upload the files here, let us take a look what changes you've made..
Since you prolly have no idea what you just did, the thing you thought were right prolly wrong.

No one else seem to have this issue from your server.
The only thing can be concluded here is fault at your end, not openkore.

**Comment by r421** [comment]
Why i get this error?
I already pull latest code from openkore.

![image](https://user-images.githubusercontent.com/26808551/32990862-dc6e9b74-cd6b-11e7-9b24-ea3a3a53286c.png)


TQ

**Comment by ncplay91** [comment]
[config.txt](https://github.com/OpenKore/openkore/files/1489103/config.txt)
[items_control.txt](https://github.com/OpenKore/openkore/files/1489104/items_control.txt)

Ok.. so with this config and items_control.txt, the sequence that happens is that.. Kore will respawn at 49% weight, get awakening potions from kafra, and then sell everything in my inventory except the awakening potions... this is weird..

**Comment by fadreus** [comment]
Rather than sell all, you should store all and set only sell things you don't need.

**Comment by ncplay91** [comment]
uhh... yeah... so you too dont see anything wrong with it, right..?
how the heck am I experiencing this weird issue?

**Comment by fadreus** [comment]
> you too dont see anything wrong with it

- Afaik you dont use `tab`. Use `space`.
- Default should be `all 0 1 0`
- In some cases, item ID/Name mismatch from openkore database (_E.g. Savage Babe & Hunter Fly monster in IRO is not match even using mob ID, unless it fall under `all`, specifically targeting this won't kill any savage babe/hunter fly until judgement day. Welp even judgement day come it wont after all)._
- If you set `all 0 0 1` I  bet it didn't match openkore DB and it proceed to sell all those not mentioned in `items_control.txt`.

**Comment by ncplay91** [comment]
I definitely gotta give a shot for your advice after I've got the time.. Just got my account banned.. LAWL!!! thanks a lot man! will be updating soon.

**Comment by ncplay91** [comment]
Ah! so that's probably why!? imma give it a shot check it now..

**Comment by sctnightcore** [comment]
@ncplay91  download new openkore , I think work XD

**Comment by ncplay91** [comment]
It doesnt work!! Oh no I sold my valuable items again!!! How the heck is this happening to me!?

**Comment by sctnightcore** [comment]
@ncplay91 
Config.txt
> sellAuto_npc_steps
http://openkore.com/index.php/SellAuto_npc_steps


items_control.txt
> all 0 1 0
http://openkore.com/index.php/Items_control.txt

**Comment by lututui** [comment]
Why the heck are you using tabs?

**Comment by supermankesasar** [comment]
try to leave item_control.txt just with one setting 

all 0 1 0

try this and see... if this one is ok, then ur item_control is broken.
from there, u can set ur own sellauto item, and store item.

**Comment by myself379** [comment]
Sometimes I feel you as a end user and its worth to take a rest for few moments. There is one time that I get frustrated why the heck my Wizard keeps killing Ant Egg when I explicity told it in mon_control.txt that 
Ant Egg 0 0 0
Monster Ant Egg 0 0 0
Andre Egg 0 0 0

and the bot still keep killing Ant Egg, I even copy paste from the console. Then after few rests (I know it's my problem, not bug; my hunch tells me so). Then I decided to read the lines 1 by 1, in the end I found out at the last line, there this:

Ant Egg 1 0 0

no wonder, my bot keeps killing Ant Egg! Waste the time! There is one time also that I have many profiles, few for Wild Rose, few is support Priest. I keep wondering why the heck the priest keep gather items when I told him not to, and keep changing the config.txt again and again. After few rests, and step-by-step trace back, then only I understand all this while I'm changing my Wild Rose bot profile. You know sometimes 

So @ncplay91 

I wondered, why your item_control.txt have tabs in it? I open in Notepad++ and turn on "Show All Character". 

At line 46 Skelworker Card have tabs
At line 48 Andre Card have tabs,
At line 77 Cyfar have tabs, 
At line 87 Honey have tabs,

I would suggest you try Red Pot (since you bot keep selling everything). and make sure you're edit the correct profile on the bot. Delete every item listed in items_control.txt to and set

all 0 0 1
Red Potion 5 1 0
